src/ui/routes/package/deploy/+page.svelte

```diff
@@ -0,0 +1,70 @@
+<!-- 
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+ -->
+<script lang="ts">
+	import { goto } from '$app/navigation';
+	import { page } from '$app/stores';
+	import { Packages } from '$lib/api';
+	import { pkgStore } from '$lib/store';
+	import { PackageErrNotFound, Spinner } from '$lib/components';
+
+	enum LoadingStatus {
+		Loading,
+		Success,
+		Error
+	}
+	let status: LoadingStatus = LoadingStatus.Loading;
+	let errMessage: string = '';
+	const pkgPath = $page.url.searchParams.get('path');
+	let pkgName: string;
+
+	if (pkgPath === null) {
+		errMessage = 'No package path provided';
+		status = LoadingStatus.Error;
+	} else {
+		pkgName = encodeURIComponent(
+			pkgPath
```

Comment:
this whole thing is super janky, would much rather see this an actual regex (what it's pretending to be rn)